File: 1-Two Sum/solution.js
function twoSum(nums, target) {
    const previousValues = {};
    for (let index = 0; index < nums.length; index++) {
        const currentNumber = nums[index];
        const neededValue = target - currentNumber;
        const index2 = previousValues[neededValue];
        if (index2 != null) {
            return [index2, index]
        } else {
            previousValues[currentNumber] = index;
        }
    }
}

console.log('Solution:', twoSum([2, 7, 11, 15], 9));

File: 1-Two Sum/solution.go
package main

import "fmt"

func main() {
	array := []int{15, 11, 2, 7}
	fmt.Println(twoSum(array, 9))
}

func twoSum(nums []int, target int) [2]int {
	for index := 0; index < len(nums); index++ {
		for index2 := 0; index2 < len(nums); index2++ {
			if nums[index] + nums[index2] == target {
				return [2]int{index, index2}
			}
		}
		nums = nums[index:]
	}
	return [2]int{0, 1}
}